# tara-cyc-hs18

## Friday, October 12

Decided to use Shiny

Samples taken from around the world
Filtered to isolate viral component
Underwent metagenomic sequencing
Mapped reads to pathways - get mapping of pathways in different areas of ocean
rpkm (reads per kilobase of transcript per million mapped reads) - normalized count of how many reads mapped to a transcript -  values - value of expression level of a pathway
5 different depths for sampling - in meters
lat long

Mapping of metagenomic data:


Website - Things to Include:
Map - Use lat and long to map the 
Leaflet - R package for maps - allows zooming functions
Want query functions - by location
If we can get dates - can add info on the temperature at the time of collection
Map - want filters for depth
Can pull up data for a sample/area - then filter by depth - show pathway information
Analysis tab to compare different regions
Link out to pathway information on MetaCyc

gutcyc.com - similar initiative - gut microbiome

Task Divy Up:
Analysis: Javier, Heather, Jasmine
Map: Agan, Dan, Kristen, Olga, Arjun
Reading: Kristen
Shiny: Ogan, Olga
Integration: Arjun
